tests/testthat/test-bind.R

```diff
@@ -930,6 +931,43 @@ test_that("vec_rbind() fallback works with tibbles", {
   expect_identical(vec_rbind(tib, df), exp)
 })
 
+test_that("vec_rbind() zaps names when name-spec is zap() and names-to is NULL", {
+  expect_identical(
+    vec_rbind(foo = c(x = 1), .names_to = NULL, .name_spec = zap()),
+    data.frame(x = 1)
+  )
+})
+
+test_that("can't zap names when `.names_to` is supplied", {
+  expect_identical(
+    vec_rbind(foo = c(x = 1), .names_to = zap(), .name_spec = zap()),
+    data.frame(x = 1)
+  )
+  expect_error(
+    vec_rbind(foo = c(x = 1), .names_to = "id", .name_spec = zap()),
+    "Can't zap outer names when `.names_to` is supplied.",
```

Comment:
I wonder if this is too strict. I almost thing that this should result in an integer `id` column (i.e. all names were zapped, which is equivalent to no names being present to start with).

Reply:
I'd prefer to restrict things for now. This zapping scheme is complicated. Currently `.name_spec = zap()` ignores all names, including inner names in the recursive case (data frames, df-cols and their columns). This doesn't seem completely right to me. Though this can now be worked around by supplying `~ NULL` instead, which is only applied to outer names.